addons/website_crm_partner_assign_commissions/models/account_invoice.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+from odoo.tools import float_compare
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    reseller_id = fields.Many2one('res.partner', 'Reseller', help='This field is used to track the reseller in order to generate commisions')
+    purchase_order_line_id = fields.Many2one('purchase.order.line', 'Reseller Purchase Order line')
+
+    def _make_commissions(self):
+        """
+            Adds a line to a purchase order for reseller_id containing the due commissions for the invoice
+            The line is only added the first time we call this method (called on reconcile)
+        """
+        self.ensure_one()
+
+        if not self.reseller_id or self.purchase_order_line_id:
+            return
+
+        total = 0
```

Comment:
This should be moved inside the loop, otherwise the value is never reset inside the loop and the total gets bigger and bigger.

Reply:
The total is computed once per invoice, so that it will create only one purchase line per invoice